broadinstitute/consent-ontology
# Accommodate DUO ontology import

## Addresses:
https://broadinstitute.atlassian.net/browse/GAWB-2523
Requires: https://github.com/broadinstitute/consent/pull/244
Requires: https://github.com/broadinstitute/consent-data-use/pull/6

## Changes:
* Upgrade the owl api to handle direct ontology imports
* Update the app so that it doesn't fail on startup if it cannot find ontology files in GCS. Instead, log an error and re-check before trying to use any GCS ontology file.
* Use DUO terms where required instead of DUOS terms. 

## Testing:
* See the testing section in https://github.com/broadinstitute/consent/pull/244
* After updating the elastic search index with https://github.com/broadinstitute/consent/pull/244, the auto-complete feature should find DUO terms such as `methods research` and `non profit` as well as DUOS terms such as `male` and `female`
* Matching should still work even with DUO terms. In addition to current unit tests (which do use some DUO terms), here are some examples:
```
DUO_0000015 --> methods research
DOID_4      --> disease
Should succeed:
{
	"purpose":{"type":"named","name":"http://purl.obolibrary.org/obo/DUO_0000015"},
	"consent":{"type":"everything"}
}
Should succeed:
{
	"purpose":{"type":"named","name":"http://purl.obolibrary.org/obo/DUO_0000015"},
	"consent":{"type":"named","name":"http://purl.obolibrary.org/obo/DUO_0000015"}
}
Should fail:
{
	"purpose":{"type":"named","name":"http://purl.obolibrary.org/obo/DOID_4"},
	"consent":{"type":"named","name":"http://purl.obolibrary.org/obo/DUO_0000015"}
}
```

---

  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
